src/codegen_common.rs

```diff
@@ -70,6 +70,7 @@ pub trait CodeGen: Serialize {
     fn get_maps(&mut self);
     fn make_struct_filter_blocks(&mut self);
     fn make_attr_filter_blocks(&mut self);
+    fn make_trace_rpc_value(&mut self);
```

Comment:
What is the difference between the two other functions below? It looks like we can merge some of this.

Reply:
make_rpc_value_from_trace aliases trace as the root node, so that you get the root node's information as the trace to return.
make_rpc_value_from_target gets the information of any node in your target graph to return.
make_trace_rpc_value allows you to return the entire trace.